src/thorlabs_mff_fastcs/controllers.py

```diff
@@ -81,16 +134,26 @@ class ThorlabsMFFHandlerR:
     response_size: int
     response_handler: Callable
     update_period: float = 0.2
+    cache: ResponseCache | None = None
 
     async def update(
         self,
         controller: ThorlabsMFF,
         attr: AttrR,
     ) -> None:
-        response = await controller.conn.send_query(
-            self.cmd(),
-            self.response_size,
-        )
+        if self.cache is not None:
+            response = await self.cache.get_response(
+                self.update_period,
+                controller.conn.send_query,
+                self.cmd(),
+                self.response_size,
+            )
```

Comment:
You could call send query with args here and pass the response awaitable in. Might be a bit neater. You could even construct it once and then either pass it in to the cache or await it directly.

Reply:
Discussed -> RunTime error solved by calling `to_await.close()`